src/libecalc/common/utils/rates.py

```diff
@@ -1038,6 +1038,11 @@ def resample(
         if freq is Frequency.NONE:
             return self.model_copy()
 
+        # Resampling everything requires resampling of e.g. stage results, inlet/outlet stream conditions.
+        # These include rates per hour, not only per day. Need to check this and convert to per day:
+        if self.unit in [Unit.KILO_PER_HOUR, Unit.ACTUAL_VOLUMETRIC_M3_PER_HOUR]:
+            self.values = [v * 24 for v in self.values]
```

Comment:
This should probably be handled by using the `to_unit` function in TimeSeries.

Can we do the same below for calendar_day_volumes (and stream_day_columes)? `self.to_calendar_day().to_volumes().cumulative().resample().to_volumes()`

Do we have any tests for these functions?

Reply:
Not many tests as I am aware of. We have one in test_rates: `test_compute_stream_day_rate` (`Rates.to_stream_day()`).

Reply:
Changed, unit check not necessary.